src/validation.cpp

```diff
@@ -3203,7 +3217,7 @@ static bool FindBlockPos(FlatFilePos &pos, unsigned int nAddSize, unsigned int n
 
     bool finalize_undo = false;
     if (!fKnown) {
-        while (vinfoBlockFile[nFile].nSize + nAddSize >= MAX_BLOCKFILE_SIZE) {
+        while (vinfoBlockFile[nFile].nSize + nAddSize >= (gArgs.GetBoolArg("-fastprune", false) ? 0x10000 /* 64kb */ : MAX_BLOCKFILE_SIZE)) {
```

Comment:
The numeric constant could be extracted into `src/validation.h` (same file where `MAX_BLOCKFILE_SIZE` is defined).

Reply:
I thought about it but since its a debug only parameter I rather keep that inline.